### PR TITLE
devcenter events preview

### PR DESCRIPTION
**Pull request by harikakotipalli**
Adding preview for events (ONlY FOR INDUSTRY EVENTS)

### CODE REVIEW

File: src/api-requests/get-industry-events.ts [hunk 1]
```diff
@@ -116,3 +116,21 @@ export const getIndustryEventBySlugFromAPI = async (
 
     return data.industryEvents.length > 0 ? data.industryEvents[0] : null;
 };
+
+export const getDraftEventFromAPI = async (
+    client: UnderlyingClient<'ApolloREST'>,
+    calculatedSlug: string
+): Promise<IndustryEvent | null> => {
+    calculatedSlug = '"' + calculatedSlug + '"';
```
Comment (justinmckenzie) [inline src/api-requests/get-industry-events.ts:124]: what is it you're trying to do here?

File: src/pages/preview/[...slug].tsx [hunk 2]
```diff
@@ -43,8 +46,16 @@ export default ContentPage;
...
-    const contents: ContentItem[] = await getPreviewContent('/' + slugString);
-    const contentItem = contents.filter(c => c.slug === slugString)[0];
+    let contentItem;
+    if (slugString.startsWith('events/')) {
+        contentItem = await getPreviewContentForEvents('/' + slugString);
+    } else {
+        const contents: ContentItem[] = await getPreviewContentForArticles(
+            '/' + slugString
+        );
+        contentItem = contents.filter(c => c.slug === slugString)[0];
```
Comment (justinmckenzie) [inline src/pages/preview/[...slug].tsx:56]: its better to use `.find` here, it returns the entire item that matches the condition.

File: src/service/get-preview-content.ts [hunk 3]
```diff
@@ -13,3 +17,13 @@ export const getPreviewContent: (
...
 };
+
+export const getPreviewContentForEvents: (
+    calculatedSlug: string
+) => Promise<ContentItem | null> = async calculatedSlug => {
+    const event = await getDraftEventFromAPI(STRAPI_CLIENT, calculatedSlug);
+    if (!event) {
+        return null;
+    }
+    return mapIndustryEventToContentItem(event);
```
Comment (justinmckenzie) [inline src/service/get-preview-content.ts:28]: `return event ? mapIndustryEventToContentItem(event) : null`

File: src/api-requests/get-industry-events.ts (hunk 1 above)
Comment (harikakotipalli) [reply src/api-requests/get-industry-events.ts:124]: have to construct Graphql args like this - add quotes to the calculated slug.

File: src/pages/preview/[...slug].tsx (hunk 2 above)
Comment (harikakotipalli) [reply src/pages/preview/[...slug].tsx:56]: old code. dint touch this, will change.

File: src/api-requests/get-industry-events.ts (hunk 1 above)
Comment (justinmckenzie) [reply src/api-requests/get-industry-events.ts:124]: ok, i was just confused by the single quote, double quote, single quote pattern. I would say just put quotes around your `calculatedSlug` template literal on line 127:

`"${calculatedSlug}"`

**Review by justinmckenzie — APPROVED** [review]

**Review by mcolella14 — APPROVED** [review]

**Pull request merged**